### PR TITLE
 Add add_column_relationship method to single table metadata

### LINKED ISSUE #1699: Add add_column_relationship method to single table metadata
### Problem Description

_As a user, I'd like to be able to specify when certain columns are related (eg. address columns) so that the SDV can handle them appropriately._

### Expected behavior

Add a new method to the metadata classes called `add_column_relationship`.

- Single table
    - `add_column_relationship(type: str, column_names: list)`
        - `type`: A string with the type of relationship. Only possibility for now is "address"
        - `column_names`: A list of column names

```python
metadata.add_column_relationship(
  type="address",
  table_name="vendors",
  column_names=["vendor_city", "vendor_state", "vendor_zipcode", ...]
)
```
Calling this method should add the relationship to the metadata so tat inspecting the metadata yields
```python
{
  "columns": {
    "user_lat": { "sdtype": "latitude" },
    "user_lon": { "sdtype": "longitude" },
  },
  "column_relationships": [{
    "type": "gps_coordinate",
    "column_names": ["user_lat", "user_lon" ],
  }]
}
```

- Multi table
    -  `add_column_relationship(type: str, table_name: str, column_names: list)`
        - `type`: A string with the type of relationship. Only possibility for now is "address"
        - `table_name`: Name of the table that the relationship is in.
        - `column_names`: A list of column names

Calling this method should add the relationship to the metadata so tat inspecting the metadata yields
```python
{
  "tables": {
    "users": {
      "columns": {
         "user_lat": { "sdtype": "latitude" },
         "user_lon": { "sdtype": "longitude" },
      },
      "column_relationships": [{
         "type": "gps_coordinate",
         "column_names": ["user_lat", "user_lon" ],
      }]
    }
  },
  "relationships": [{
    ...
  }]
}
```
### Additional context

- This should use the same validation as #1698

### PR DESCRIPTION
**Pull request by frances-h**
CU-86aytvu71
Resolve #1699

### CODE REVIEW

**Comment by sdv-team** [comment]
Task linked: [CU-86aytvu71 SDV - Add add_column_relationship method to single table metadata #1699](https://app.clickup.com/t/86aytvu71)

File: tests/integration/metadata/test_multi_table.py [hunk 1]
```diff
@@ -25,6 +25,23 @@ def test_multi_table_metadata():
...
+    _, instance = download_demo('multi_table', 'fake_hotels')
+    instance.update_column('hotels', 'city', sdtype='city')
+    instance.update_column('hotels', 'state', sdtype='state')
+
+    # Run
+    instance.add_column_relationship('address', 'hotels', ['city', 'state'])
+
+    # Assert
+    instance.validate()
+    instance.tables['hotels'].column_relationships = [
```
Comment (fealho) [inline tests/integration/metadata/test_multi_table.py:40]: Add assert.

**Review by fealho — COMMENTED** [review]
Could you also add an integration test for the single table as well?

**Review by fealho — APPROVED** [review]
👍

File: tests/unit/metadata/test_multi_table.py [hunk 2]
```diff
@@ -734,28 +734,58 @@ def test_remove_relationship_relationship_not_found(self, warnings_mock):
...
         err_msg = (
             "Cannot use foreign keys {'owner'} in column relationship."
         )
         with pytest.raises(InvalidMetadataError, match=err_msg):
-            instance._validate_column_relationships_foreign_keys(table_accounts, ['owner'])
+            instance._validate_column_relationships_foreign_keys(column_relationships, ['owner'])
+
+    def test_add_column_relationship(self):
+        """Test ``add_column_relationship`` adds a column relationship."""
+        instance = MultiTableMetadata()
```
Comment (R-Palazzo) [inline tests/unit/metadata/test_multi_table.py:751]: Add `# Setup`

**Review by R-Palazzo — APPROVED** [review]
LGTM!

**Pull request merged**